### DISCUSSION TITLE
wix.exe : error WIX0001: System.IO.IOException: The pipe has been ended

### DISCUSSION THREAD

**Discussion by sumedhbhogle** [wixtoolset/issues#7904] [Questions] [answered]
Hello Team,

I had issues with WiX3.0 where heat tool was not able to harvest paths greater than 260length of chars. 

`heat.exe : error HEAT0001 : The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.`

I switched to WiX4.0 where issue is resolved, I am able to harvest path length more than 260. But now I am getting this wix.exe error, it looks like some internal communication issue (pipe ended). But I am not able to diagnose the actual issue. Can you please help me to diagnose this further?

`wix.exe : error WIX0001: System.IO.IOException: The pipe has been ended`

**Comment by sumedhbhogle** [comment, marked as answer]
This turns out to be [long path](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry) issue on windows. 

To resolve this in wix project ensure that path you pass as BindPath is prefixed with "\\?\" as mentioned in above msdn article. Also make sure that registry settings are applied as mentioned in document.

`   <ItemGroup>
    <HarvestDirectory Include="$(GIT_WORKSPACE)\build\[long_path]">
      <ComponentGroupName>COMPONENT_NAME</ComponentGroupName>
      <DirectoryRefId>TARGETDIR</DirectoryRefId>
    </HarvestDirectory>
    <BindPath Include="\\?\$(GIT_WORKSPACE)\[long_path]" />
  </ItemGroup>`

GIT_WORKSPACE is defined in Directory.Build.props as described in [MSBuild Wix](https://wixtoolset.org/docs/tools/msbuild/) document.

Thanks!

**Reply by McGeifer** [reply]
Thanks for that! Took me three days to figure this out/ find this post.

**Reply by robmen** [reply]
Note, WiX v7 should generally handle long file names better.

**Reply by McGeifer** [reply]
> Note, WiX v7 should generally handle long file names better.

Good to know, thanks.
